engines/sci/graphics/frameout.cpp

```diff
@@ -55,31 +57,37 @@
 
 namespace Sci {
 
-GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxScreen *screen, GfxPalette32 *palette, GfxTransitions32 *transitions) :
+GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxPalette32 *palette, GfxTransitions32 *transitions, GfxCursor32 *cursor) :
 	_isHiRes(ConfMan.getBool("enable_high_resolution_graphics")),
 	_palette(palette),
+	_cursor(cursor),
 	_resMan(resMan),
-	_screen(screen),
 	_segMan(segMan),
 	_transitions(transitions),
 	_benchmarkingFinished(false),
 	_throttleFrameOut(true),
 	_throttleState(0),
-	// TODO: Stop using _gfxScreen
-	_currentBuffer(screen->getDisplayWidth(), screen->getDisplayHeight(), nullptr),
 	_remapOccurred(false),
 	_frameNowVisible(false),
-	_screenRect(screen->getDisplayWidth(), screen->getDisplayHeight()),
 	_overdrawThreshold(0),
 	_palMorphIsOn(false) {
 
-	_currentBuffer.setPixels(calloc(1, screen->getDisplayWidth() * screen->getDisplayHeight()));
-
-	// QFG4 is the only SCI32 game that doesn't have a high-resolution toggle
+	// QFG4 is the only SCI32 game that doesn't have a high-resolution version
 	if (g_sci->getGameId() == GID_QFG4) {
 		_isHiRes = false;
 	}
 
+	if (g_sci->getGameId() == GID_PHANTASMAGORIA) {
+		_currentBuffer = Buffer(630, 450, nullptr);
+	} else if (_isHiRes) {
+		_currentBuffer = Buffer(640, 480, nullptr);
+	} else {
+		_currentBuffer = Buffer(320, 200, nullptr);
+	}
```

Comment:
I just see that you changed some of that kPlatform code, when you implemented ef2c44bf1f2cf128a20096b3bcca2578634814b1. And your code basically breaks how King's Quest 6 worked.

King's Quest 6 was supposed to be hi-res when "force hi-res" is enabled OR platform is set to Windows. And now the flag is read directly, which is not only incorrect, but platform being Windows is fully ignored now.

Possibly other games may be affected too by those changes, because now all sorts of games get bad answers on those kPlatform calls.

So I guess I will not touch it (otherwise I break something in SCI32), but will instead ask you to please fix it.